src/CONST.ts

```diff
@@ -2970,6 +2970,10 @@ const CONST = {
             return new RegExp(`[\\n\\s]|${this.SPECIAL_CHAR.source}|${this.EMOJI.source}`, 'gu');
         },
 
+        get ALL_EMOJIS() {
+            return new RegExp(this.EMOJIS, this.EMOJIS.flags.concat('g'));
```

Comment:
Do we even need this, or could we directly add the `g` flag in `EMOJIS` above? As far as I can tell, the only place we use `EMOJIS` is in `EmojiUtils.splitTextWithEmojis`, and we also add `g` flag there.

Reply:
@francoisl  I think the answer to your question is here:
https://github.com/Expensify/App/pull/40692#discussion_r1580907658

Reply:
Got it, thanks.
Is there any situation where we'll want to use `CONST.REGEX.EMOJIS` without the `g` flag? If not, let's at least add the comment about cloning that's currently in `splitTextWithEmojis()` in here, to add context to other people that read it and wouldn't know if they need to use `EMOJIS` or `ALL_EMOJIS`.
Ideally though, we could move it to a more private scope that's not available outside of this file, so that someone can't accidentally use `CONST.REGEX.EMOJIS`.